tiiuae/ghaf
# Shared memory support for NVIDIA Jetson Orin: engineering build

## Description of changes



## Checklist for things done



- [ X] Summary of the proposed changes in the PR description



## Testing



Overall scenario:
- login to two test virtual machines
- run the memsocket demon and the memtest program of both sides
- verify results

STEPS:

On the NVIDIA console open two windows. 

In the first window run the following commands:

\# Login from host to netvm, as test VMs are unavailable from host
ssh 192.168.101.1
\# login to the first vm:
ssh test1-vm.ghaf
\#start shared memory - socket demon and the test application listener
memsocket -c ./test.sock &
memtest ./test.sock


\------------------------ terminal output ------------------------

`
[ghaf@ghaf-host:\~]$ ssh 192.168.101.1
(ghaf@192.168.101.1) Password: 
Last login: Fri Jul 28 09:38:54 2023 from 192.168.101.2
[ghaf@net-vm:\~]$ ssh test1-vm.ghaf
(ghaf@test1-vm.ghaf) Password: 
Last login: Fri Jul 28 09:31:38 2023 from 192.168.100.1

[ghaf@test1-vm:\~]$ memsocket -c ./test.sock &
[1] 698

[ghaf@test1-vm:\~]$ memtest ./test.sock
Waiting for a connection.
`
\------------------------------------------------------------------------

In the second window run the following commands:

\# Login from host to netvm, as test VMs are unavailable from host
ssh 192.168.101.1
\# login to the second vm:
ssh test2-vm.ghaf
\#start shared memory - socket demon and the test application sender
memsocket -s ./test.sock 3 &
memtest ./test.sock /dev/random 100M
#verify results on both windows 
`
[ghaf@ghaf-host:\~]$ ssh 192.168.101.1
(ghaf@192.168.101.1) Password: 
Last login: Fri Jul 28 09:38:54 2023 from 192.168.101.2
[ghaf@net-vm:\~]$ ssh test2-vm.ghaf
(ghaf@test2-vm.ghaf) Password: 
Last login: Fri Jul 28 09:41:05 2023 from 192.168.100.1
[ghaf@test2-vm:\~]$ memsocket -s ./test.sock 3 &
[1] 721

[ghaf@test2-vm:\~]$ memtest ./test.sock /dev/random 100M
Sent/received 104857601 bytes 8 Mbytes/sec
real 12.230s
user 0.380s
sys  0.450s

[ghaf@test2-vm:~]$ 

`
